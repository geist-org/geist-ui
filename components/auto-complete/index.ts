import AutoComplete from './auto-complete'
import AutoCompleteItem from './auto-complete-item'
import AutoCompleteSearching from './auto-complete-searching'
import AutoCompleteEmpty from './auto-complete-empty'

export type AutoCompleteComponentType = typeof AutoComplete & {
  Item: typeof AutoCompleteItem
  Option: typeof AutoCompleteItem
  Searching: typeof AutoCompleteSearching
  Empty: typeof AutoCompleteEmpty
}
;(AutoComplete as AutoCompleteComponentType).Item = AutoCompleteItem
;(AutoComplete as AutoCompleteComponentType).Option = AutoCompleteItem
;(AutoComplete as AutoCompleteComponentType).Searching = AutoCompleteSearching
;(AutoComplete as AutoCompleteComponentType).Empty = AutoCompleteEmpty

export type {
  AutoCompleteOption,
  AutoCompleteOptions,
  AutoCompleteProps,
  AutoCompleteTypes,
} from './auto-complete'
export default AutoComplete as AutoCompleteComponentType
