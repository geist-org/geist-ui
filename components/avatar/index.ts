import Avatar from './avatar'
import AvatarGroup from './avatar-group'

export type AvatarComponentType = typeof Avatar & {
  Group: typeof AvatarGroup
}
;(Avatar as AvatarComponentType).Group = AvatarGroup

export type { AvatarProps } from './avatar'
export type { AvatarGroupProps } from './avatar-group'

export default Avatar as AvatarComponentType
