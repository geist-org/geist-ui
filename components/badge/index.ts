import Badge from './badge'
import BadgeAnchor from './badge-anchor'

export type BadgeComponentType = typeof Badge & {
  Anchor: typeof BadgeAnchor
}
;(Badge as BadgeComponentType).Anchor = BadgeAnchor

export type { BadgeProps, BadgeTypes } from './badge'
export type { BadgeAnchorProps, BadgeAnchorPlacement } from './badge-anchor'
export default Badge as BadgeComponentType
