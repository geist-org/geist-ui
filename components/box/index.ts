import Box from './box'

export type { BoxProps } from './box'
export default Box
