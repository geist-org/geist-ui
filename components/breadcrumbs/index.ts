import Breadcrumbs from './breadcrumbs'
import BreadcrumbsItem from './breadcrumbs-item'
import BreadcrumbsSeparator from './breadcrumbs-separator'

export type BreadcrumbsComponentType = typeof Breadcrumbs & {
  Item: typeof BreadcrumbsItem
  Separator: typeof BreadcrumbsSeparator
}
;(Breadcrumbs as BreadcrumbsComponentType).Item = BreadcrumbsItem
;(Breadcrumbs as BreadcrumbsComponentType).Separator = BreadcrumbsSeparator

export type { BreadcrumbsProps } from './breadcrumbs'
export type { BreadcrumbsItemProps } from './breadcrumbs-item'
export type { BreadcrumbsSeparatorProps } from './breadcrumbs-separator'
export default Breadcrumbs as BreadcrumbsComponentType
