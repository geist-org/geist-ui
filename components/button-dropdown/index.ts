import ButtonDropdown from './button-dropdown'
import ButtonDropdownItem from './button-dropdown-item'

type ButtonDropdownType = typeof ButtonDropdown & {
  Item: typeof ButtonDropdownItem
}
;(ButtonDropdown as ButtonDropdownType).Item = ButtonDropdownItem

export type { ButtonDropdownProps, ButtonDropdownTypes } from './button-dropdown'
export type {
  ButtonDropdownItemProps,
  ButtonDropdownItemTypes,
} from './button-dropdown-item'
export default ButtonDropdown as ButtonDropdownType
