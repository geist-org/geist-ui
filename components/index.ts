// @ts-ignore
/// <reference types="styled-jsx" />

export { default as AutoComplete } from './auto-complete'
export type { AutoCompleteProps } from './auto-complete'

export { default as Avatar } from './avatar'
export type { AvatarProps, AvatarGroupProps } from './avatar'

export { default as Badge } from './badge'
export type { BadgeProps, BadgeAnchorProps } from './badge'

export { default as Breadcrumbs } from './breadcrumbs'
export type {
  BreadcrumbsProps,
  BreadcrumbsItemProps,
  BreadcrumbsSeparatorProps,
} from './breadcrumbs'

export { default as Button } from './button'
export type { ButtonProps } from './button'

export { default as ButtonDropdown } from './button-dropdown'
export type { ButtonDropdownProps, ButtonDropdownItemProps } from './button-dropdown'

export { default as ButtonGroup } from './button-group'
export type { ButtonGroupProps } from './button-group'

export { default as Capacity } from './capacity'
export type { CapacityProps } from './capacity'

export { default as Card } from './card'
export type { CardProps, CardContentProps, CardFooterProps } from './card'

export { default as Checkbox } from './checkbox'
export type { CheckboxProps, CheckboxGroupProps } from './checkbox'

export { default as Code } from './code'
export type { CodeProps } from './code'

export { default as Collapse } from './collapse'
export type { CollapseProps, CollapseGroupProps } from './collapse'

export { default as Description } from './description'
export type { DescriptionProps } from './description'

export { default as Display } from './display'
export type { DisplayProps } from './display'

export { default as Divider } from './divider'
export type { DividerProps } from './divider'

export { default as Dot } from './dot'
export type { DotProps } from './dot'

export { default as Drawer } from './drawer'
export type { DrawerProps } from './drawer'

export { default as Fieldset } from './fieldset'
export type {
  FieldsetProps,
  FieldsetTitleProps,
  FieldsetSubtitleProps,
  FieldsetGroupProps,
  FieldsetFooterProps,
  FieldsetContentProps,
} from './fieldset'

export { default as GeistProvider } from './geist-provider'
export type { GeistProviderProps } from './geist-provider'

export { default as Grid } from './grid'
export type { GridProps, GridContainerProps } from './grid'

export { default as Image } from './image'
export type { ImageProps, ImageBrowserProps } from './image'

export { default as Input } from './input'
export type { InputProps, InputPasswordProps } from './input'

export { default as Keyboard } from './keyboard'
export type { KeyboardProps } from './keyboard'

export { default as Link } from './link'
export type { LinkProps } from './link'

export { default as Loading } from './loading'
export type { LoadingProps } from './loading'

export { default as Modal } from './modal'
export type {
  ModalProps,
  ModalTitleProps,
  ModalSubtitleProps,
  ModalContentProps,
  ModalActionProps,
} from './modal'

export { default as Note } from './note'
export type { NoteProps } from './note'

export { default as Page } from './page'
export type {
  PageProps,
  PageHeaderProps,
  PageContentProps,
  PageFooterProps,
} from './page'

export { default as Pagination } from './pagination'
export type {
  PaginationProps,
  PaginationNextProps,
  PaginationPreviousProps,
} from './pagination'

export { default as Popover } from './popover'
export type { PopoverProps, PopoverItemProps } from './popover'

export { default as Progress } from './progress'
export type { ProgressProps } from './progress'

export { default as Radio } from './radio'
export type { RadioProps, RadioGroupProps, RadioDescriptionProps } from './radio'

export { default as Rating } from './rating'
export type { RatingProps } from './rating'

export { default as Select } from './select'
export type { SelectProps, SelectOptionProps } from './select'

export { default as Slider } from './slider'
export type { SliderProps } from './slider'

export { default as Snippet } from './snippet'
export type { SnippetProps } from './snippet'

export { default as Spacer } from './spacer'
export type { SpacerProps } from './spacer'

export { default as Spinner } from './spinner'
export type { SpinnerProps } from './spinner'

export { default as Table } from './table'
export type { TableProps, TableColumnProps } from './table'

export { default as Tabs } from './tabs'
export type { TabsProps } from './tabs'

export { default as Tag } from './tag'
export type { TagProps } from './tag'

export { default as Text } from './text'
export type { TextProps } from './text'

export { default as Textarea } from './textarea'
export type { TextareaProps } from './textarea'

export { default as Themes } from './themes'
export type { GeistUIThemes, GeistUserTheme } from './themes'

export { default as Toggle } from './toggle'
export type { ToggleProps } from './toggle'

export { default as Tooltip } from './tooltip'
export type { TooltipProps } from './tooltip'

export { default as Tree } from './tree'
export type { TreeProps } from './tree'

export { default as useAllThemes } from './use-all-themes'
export type { AllThemesConfig } from './use-all-themes'

export { default as useToasts } from './use-toasts'
export type { Toast, ToastInput, ToastAction, ToastLayout } from './use-toasts'

export { default as User } from './user'
export type { UserProps } from './user'

export { default as useBodyScroll } from './use-body-scroll'
export type { BodyScrollOptions } from './use-body-scroll'

export { default as useClipboard } from './use-clipboard'
export type { UseClipboardOptions } from './use-clipboard'

export { default as useMediaQuery } from './use-media-query'
export type { ResponsiveOptions, ResponsiveBreakpoint } from './use-media-query'

export { default as useKeyboard, KeyMod, KeyCode } from './use-keyboard'
export type { KeyboardOptions, UseKeyboardHandler } from './use-keyboard'

export { default as useInput } from './use-input'
export { default as useModal } from './use-modal'
export { default as useTabs } from './use-tabs'
export { default as useClickAway } from './use-click-away'
export { default as useCurrentState } from './use-current-state'
export { default as CssBaseline } from './css-baseline'
export { default as useTheme } from './use-theme'
export { default as useClasses } from './use-classes'
export { default as useScale } from './use-scale'
export { withScale, ScalePropKeys, ScaleContext } from './use-scale'
export type {
  ScaleProps,
  ScaleConfig,
  GetScalePropsFunction,
  GetAllScalePropsFunction,
} from './use-scale'

export { useRect } from './utils/layouts'
export type { ReactiveDomReact } from './utils/layouts'
export { default as Highlight } from './shared/highlight'
export type { HighlightProps } from './shared/highlight'
