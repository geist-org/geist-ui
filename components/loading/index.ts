import Loading from './loading'

export type { LoadingProps, LoadingTypes } from './loading'
export default Loading
