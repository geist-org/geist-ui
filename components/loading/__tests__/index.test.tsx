import React from 'react'
import { mount } from 'enzyme'
import { Loading } from 'components'

describe('Loading', () => {
  it('should render correctly', () => {
    const wrapper = mount(<Loading />)
    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).toMatchSnapshot()
  })

  it('should work with different types', () => {
    const wrapper = mount(
      <div>
        <Loading type="success" />
        <Loading type="secondary" />
        <Loading type="warning" />
        <Loading type="error" />
      </div>,
    )
    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).toMatchSnapshot()
  })

  it('should work with custom styles', () => {
    const wrapper = mount(
      <div>
        <Loading color="#fff" />
        <Loading unit="20%" />
        <Loading unit="10px" />
      </div>,
    )
    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).toMatchSnapshot()
  })

  it('should work with children', () => {
    const wrapper = mount(<Loading>test-children</Loading>)
    expect(wrapper.find('.loading').text()).toContain('test-children')
  })
})
