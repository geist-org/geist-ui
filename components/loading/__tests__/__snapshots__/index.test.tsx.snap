// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should render correctly 1`] = `
"<div class=\\"loading-container\\"><span class=\\"loading\\"><i></i><i></i><i></i></span><style>
        .loading-container {
          display: inline-flex;
          align-items: center;
          position: relative;
          width: 100%;
          height: 100%;
        }

        label {
          margin-right: 8pt;
          color: #666;
        }

        label :global(*) {
          margin: 0;
        }

        .loading {
          position: absolute;
          top: 0;
          left: 0;
          right: 0;
          bottom: 0;
          width: 100%;
          height: 100%;
          display: flex;
          justify-content: center;
          align-items: center;
          background-color: transparent;
          user-select: none;
        }

        i {
          width: 4px;
          height: 4px;
          border-radius: 50%;
          background-color: #444;
          margin: 0 1px;
          display: inline-block;
          animation: loading-blink 1.4s infinite both;
        }

        i:nth-child(2) {
          animation-delay: 0.2s;
        }

        i:nth-child(3) {
          animation-delay: 0.4s;
        }

        @keyframes loading-blink {
          0% {
            opacity: 0.2;
          }

          20% {
            opacity: 1;
          }

          100% {
            opacity: 0.2;
          }
        }
      </style></div>"
`;

exports[`Loading should render correctly 2`] = `[Function]`;

exports[`Loading should work with custom styles 1`] = `
"<div><div class=\\"loading-container\\"><span class=\\"loading\\"><i></i><i></i><i></i></span><style>
        .loading-container {
          display: inline-flex;
          align-items: center;
          position: relative;
          width: 100%;
          height: 100%;
        }

        label {
          margin-right: 8pt;
          color: #666;
        }

        label :global(*) {
          margin: 0;
        }

        .loading {
          position: absolute;
          top: 0;
          left: 0;
          right: 0;
          bottom: 0;
          width: 100%;
          height: 100%;
          display: flex;
          justify-content: center;
          align-items: center;
          background-color: transparent;
          user-select: none;
        }

        i {
          width: 4px;
          height: 4px;
          border-radius: 50%;
          background-color: #fff;
          margin: 0 1px;
          display: inline-block;
          animation: loading-blink 1.4s infinite both;
        }

        i:nth-child(2) {
          animation-delay: 0.2s;
        }

        i:nth-child(3) {
          animation-delay: 0.4s;
        }

        @keyframes loading-blink {
          0% {
            opacity: 0.2;
          }

          20% {
            opacity: 1;
          }

          100% {
            opacity: 0.2;
          }
        }
      </style></div><div class=\\"loading-container\\"><span class=\\"loading\\"><i></i><i></i><i></i></span><style>
        .loading-container {
          display: inline-flex;
          align-items: center;
          position: relative;
          width: 100%;
          height: 100%;
        }

        label {
          margin-right: 8pt;
          color: #666;
        }

        label :global(*) {
          margin: 0;
        }

        .loading {
          position: absolute;
          top: 0;
          left: 0;
          right: 0;
          bottom: 0;
          width: 100%;
          height: 100%;
          display: flex;
          justify-content: center;
          align-items: center;
          background-color: transparent;
          user-select: none;
        }

        i {
          width: 4px;
          height: 4px;
          border-radius: 50%;
          background-color: #444;
          margin: 0 1px;
          display: inline-block;
          animation: loading-blink 1.4s infinite both;
        }

        i:nth-child(2) {
          animation-delay: 0.2s;
        }

        i:nth-child(3) {
          animation-delay: 0.4s;
        }

        @keyframes loading-blink {
          0% {
            opacity: 0.2;
          }

          20% {
            opacity: 1;
          }

          100% {
            opacity: 0.2;
          }
        }
      </style></div><div class=\\"loading-container\\"><span class=\\"loading\\"><i></i><i></i><i></i></span><style>
        .loading-container {
          display: inline-flex;
          align-items: center;
          position: relative;
          width: 100%;
          height: 100%;
        }

        label {
          margin-right: 8pt;
          color: #666;
        }

        label :global(*) {
          margin: 0;
        }

        .loading {
          position: absolute;
          top: 0;
          left: 0;
          right: 0;
          bottom: 0;
          width: 100%;
          height: 100%;
          display: flex;
          justify-content: center;
          align-items: center;
          background-color: transparent;
          user-select: none;
        }

        i {
          width: 4px;
          height: 4px;
          border-radius: 50%;
          background-color: #444;
          margin: 0 1px;
          display: inline-block;
          animation: loading-blink 1.4s infinite both;
        }

        i:nth-child(2) {
          animation-delay: 0.2s;
        }

        i:nth-child(3) {
          animation-delay: 0.4s;
        }

        @keyframes loading-blink {
          0% {
            opacity: 0.2;
          }

          20% {
            opacity: 1;
          }

          100% {
            opacity: 0.2;
          }
        }
      </style></div></div>"
`;

exports[`Loading should work with custom styles 2`] = `[Function]`;

exports[`Loading should work with different sizes 1`] = `
"<div><div class=\\"loading-container\\"><span class=\\"loading\\"><i></i><i></i><i></i></span><style>
        .loading-container {
          display: inline-flex;
          align-items: center;
          position: relative;
          width: 100%;
          height: 100%;
        }

        label {
          margin-right: 8pt;
          color: #666;
        }

        label :global(*) {
          margin: 0;
        }

        .loading {
          position: absolute;
          top: 0;
          left: 0;
          right: 0;
          bottom: 0;
          width: 100%;
          height: 100%;
          display: flex;
          justify-content: center;
          align-items: center;
          background-color: transparent;
          user-select: none;
        }

        i {
          width: 2px;
          height: 2px;
          border-radius: 50%;
          background-color: #444;
          margin: 0 1px;
          display: inline-block;
          animation: loading-blink 1.4s infinite both;
        }

        i:nth-child(2) {
          animation-delay: 0.2s;
        }

        i:nth-child(3) {
          animation-delay: 0.4s;
        }

        @keyframes loading-blink {
          0% {
            opacity: 0.2;
          }

          20% {
            opacity: 1;
          }

          100% {
            opacity: 0.2;
          }
        }
      </style></div><div class=\\"loading-container\\"><span class=\\"loading\\"><i></i><i></i><i></i></span><style>
        .loading-container {
          display: inline-flex;
          align-items: center;
          position: relative;
          width: 100%;
          height: 100%;
        }

        label {
          margin-right: 8pt;
          color: #666;
        }

        label :global(*) {
          margin: 0;
        }

        .loading {
          position: absolute;
          top: 0;
          left: 0;
          right: 0;
          bottom: 0;
          width: 100%;
          height: 100%;
          display: flex;
          justify-content: center;
          align-items: center;
          background-color: transparent;
          user-select: none;
        }

        i {
          width: 3px;
          height: 3px;
          border-radius: 50%;
          background-color: #444;
          margin: 0 1px;
          display: inline-block;
          animation: loading-blink 1.4s infinite both;
        }

        i:nth-child(2) {
          animation-delay: 0.2s;
        }

        i:nth-child(3) {
          animation-delay: 0.4s;
        }

        @keyframes loading-blink {
          0% {
            opacity: 0.2;
          }

          20% {
            opacity: 1;
          }

          100% {
            opacity: 0.2;
          }
        }
      </style></div><div class=\\"loading-container\\"><span class=\\"loading\\"><i></i><i></i><i></i></span><style>
        .loading-container {
          display: inline-flex;
          align-items: center;
          position: relative;
          width: 100%;
          height: 100%;
        }

        label {
          margin-right: 8pt;
          color: #666;
        }

        label :global(*) {
          margin: 0;
        }

        .loading {
          position: absolute;
          top: 0;
          left: 0;
          right: 0;
          bottom: 0;
          width: 100%;
          height: 100%;
          display: flex;
          justify-content: center;
          align-items: center;
          background-color: transparent;
          user-select: none;
        }

        i {
          width: 4px;
          height: 4px;
          border-radius: 50%;
          background-color: #444;
          margin: 0 1px;
          display: inline-block;
          animation: loading-blink 1.4s infinite both;
        }

        i:nth-child(2) {
          animation-delay: 0.2s;
        }

        i:nth-child(3) {
          animation-delay: 0.4s;
        }

        @keyframes loading-blink {
          0% {
            opacity: 0.2;
          }

          20% {
            opacity: 1;
          }

          100% {
            opacity: 0.2;
          }
        }
      </style></div><div class=\\"loading-container\\"><span class=\\"loading\\"><i></i><i></i><i></i></span><style>
        .loading-container {
          display: inline-flex;
          align-items: center;
          position: relative;
          width: 100%;
          height: 100%;
        }

        label {
          margin-right: 8pt;
          color: #666;
        }

        label :global(*) {
          margin: 0;
        }

        .loading {
          position: absolute;
          top: 0;
          left: 0;
          right: 0;
          bottom: 0;
          width: 100%;
          height: 100%;
          display: flex;
          justify-content: center;
          align-items: center;
          background-color: transparent;
          user-select: none;
        }

        i {
          width: 5px;
          height: 5px;
          border-radius: 50%;
          background-color: #444;
          margin: 0 1px;
          display: inline-block;
          animation: loading-blink 1.4s infinite both;
        }

        i:nth-child(2) {
          animation-delay: 0.2s;
        }

        i:nth-child(3) {
          animation-delay: 0.4s;
        }

        @keyframes loading-blink {
          0% {
            opacity: 0.2;
          }

          20% {
            opacity: 1;
          }

          100% {
            opacity: 0.2;
          }
        }
      </style></div></div>"
`;

exports[`Loading should work with different sizes 2`] = `[Function]`;

exports[`Loading should work with different types 1`] = `
"<div><div class=\\"loading-container\\"><span class=\\"loading\\"><i></i><i></i><i></i></span><style>
        .loading-container {
          display: inline-flex;
          align-items: center;
          position: relative;
          width: 100%;
          height: 100%;
        }

        label {
          margin-right: 8pt;
          color: #666;
        }

        label :global(*) {
          margin: 0;
        }

        .loading {
          position: absolute;
          top: 0;
          left: 0;
          right: 0;
          bottom: 0;
          width: 100%;
          height: 100%;
          display: flex;
          justify-content: center;
          align-items: center;
          background-color: transparent;
          user-select: none;
        }

        i {
          width: 4px;
          height: 4px;
          border-radius: 50%;
          background-color: #0070f3;
          margin: 0 1px;
          display: inline-block;
          animation: loading-blink 1.4s infinite both;
        }

        i:nth-child(2) {
          animation-delay: 0.2s;
        }

        i:nth-child(3) {
          animation-delay: 0.4s;
        }

        @keyframes loading-blink {
          0% {
            opacity: 0.2;
          }

          20% {
            opacity: 1;
          }

          100% {
            opacity: 0.2;
          }
        }
      </style></div><div class=\\"loading-container\\"><span class=\\"loading\\"><i></i><i></i><i></i></span><style>
        .loading-container {
          display: inline-flex;
          align-items: center;
          position: relative;
          width: 100%;
          height: 100%;
        }

        label {
          margin-right: 8pt;
          color: #666;
        }

        label :global(*) {
          margin: 0;
        }

        .loading {
          position: absolute;
          top: 0;
          left: 0;
          right: 0;
          bottom: 0;
          width: 100%;
          height: 100%;
          display: flex;
          justify-content: center;
          align-items: center;
          background-color: transparent;
          user-select: none;
        }

        i {
          width: 4px;
          height: 4px;
          border-radius: 50%;
          background-color: #666;
          margin: 0 1px;
          display: inline-block;
          animation: loading-blink 1.4s infinite both;
        }

        i:nth-child(2) {
          animation-delay: 0.2s;
        }

        i:nth-child(3) {
          animation-delay: 0.4s;
        }

        @keyframes loading-blink {
          0% {
            opacity: 0.2;
          }

          20% {
            opacity: 1;
          }

          100% {
            opacity: 0.2;
          }
        }
      </style></div><div class=\\"loading-container\\"><span class=\\"loading\\"><i></i><i></i><i></i></span><style>
        .loading-container {
          display: inline-flex;
          align-items: center;
          position: relative;
          width: 100%;
          height: 100%;
        }

        label {
          margin-right: 8pt;
          color: #666;
        }

        label :global(*) {
          margin: 0;
        }

        .loading {
          position: absolute;
          top: 0;
          left: 0;
          right: 0;
          bottom: 0;
          width: 100%;
          height: 100%;
          display: flex;
          justify-content: center;
          align-items: center;
          background-color: transparent;
          user-select: none;
        }

        i {
          width: 4px;
          height: 4px;
          border-radius: 50%;
          background-color: #f5a623;
          margin: 0 1px;
          display: inline-block;
          animation: loading-blink 1.4s infinite both;
        }

        i:nth-child(2) {
          animation-delay: 0.2s;
        }

        i:nth-child(3) {
          animation-delay: 0.4s;
        }

        @keyframes loading-blink {
          0% {
            opacity: 0.2;
          }

          20% {
            opacity: 1;
          }

          100% {
            opacity: 0.2;
          }
        }
      </style></div><div class=\\"loading-container\\"><span class=\\"loading\\"><i></i><i></i><i></i></span><style>
        .loading-container {
          display: inline-flex;
          align-items: center;
          position: relative;
          width: 100%;
          height: 100%;
        }

        label {
          margin-right: 8pt;
          color: #666;
        }

        label :global(*) {
          margin: 0;
        }

        .loading {
          position: absolute;
          top: 0;
          left: 0;
          right: 0;
          bottom: 0;
          width: 100%;
          height: 100%;
          display: flex;
          justify-content: center;
          align-items: center;
          background-color: transparent;
          user-select: none;
        }

        i {
          width: 4px;
          height: 4px;
          border-radius: 50%;
          background-color: #e00;
          margin: 0 1px;
          display: inline-block;
          animation: loading-blink 1.4s infinite both;
        }

        i:nth-child(2) {
          animation-delay: 0.2s;
        }

        i:nth-child(3) {
          animation-delay: 0.4s;
        }

        @keyframes loading-blink {
          0% {
            opacity: 0.2;
          }

          20% {
            opacity: 1;
          }

          100% {
            opacity: 0.2;
          }
        }
      </style></div></div>"
`;

exports[`Loading should work with different types 2`] = `[Function]`;
